### PR TITLE
DOC: Making public the internal omnisci documentation

### PR DESCRIPTION
**Pull request by datapythonista**
CC: @xmnlab

### CODE REVIEW

**Comment by xmnlab** [comment]
@datapythonista sorry .. I lost for some reason this notification.
I did an overview and it seems the text was not updated too much since I started this backend.
I am taking a look with more details now.

File: docs/source/user_guide/backends/omnisci.rst [hunk 1]
```diff
@@ -7,43 +17,42 @@ In this document it would be explained the main aspects of `ibis` and
 Modules
 -------
 
-omniscidb backend has 5 modules, which 3 of them are the main modules:
+omniscidb backend has 5 modules, the main three are:
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:20]: it seems now omniscidb has 7 modules

File: docs/source/user_guide/backends/omnisci.rst [hunk 2]
```diff
@@ -7,43 +17,42 @@ In this document it would be explained the main aspects of `ibis` and
...
+the follow code into `ibis.__init__`:
 
 .. code-block:: python
 
     with suppress(ImportError):
         # pip install ibis-framework[omniscidb]
         import ibis.omniscidb.api as omniscidb
 
-Basically, there is 3 function on `api` module>
+There are 3 functions in the `api` module:
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:40]: maybe you can replace `3` by `three`,  similar as you did above (https://github.com/ibis-project/ibis/pull/2265/files#diff-4ae26e229bda1f1f9f4fda583d639683R20).

File: docs/source/user_guide/backends/omnisci.rst [hunk 3]
```diff
@@ -7,43 +17,42 @@ In this document it would be explained the main aspects of `ibis` and
...
 
 .. code-block:: python
 
     t = omniscidb_cli.table('flights_2008_10k')
     proj = t['arrtime', 'arrdelay']
     print(ibis.omniscidb.compile(proj))
 
-`connect` method instantiates a `MapDClient` object that connect to the `omniscidb`
-database specified:
+`connect` method instantiates a `MapDClient` object that connect to the specified
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:54]:
```suggestion
`connect` method instantiates a `OmniSciDBClient` object that connect to the specified
```

File: docs/source/user_guide/backends/omnisci.rst [hunk 4]
```diff
@@ -80,14 +89,8 @@ Its main methods are:
...
 
-`parse` method ... # @TODO
-
-`to_ibis` method ... # @TODO
-
-`from_ibis` method ... # @TODO
-
-`MapDClient` class is used to connect to `omniscidb` database and manipulation data
-expression. Its main methods are:
+`MapDClient` class is used to connect to an `omniscidb` database and manipulate data
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:92]:
```suggestion
`OmniSciDBClient` class is used to connect to an `omniscidb` database and manipulate data
```

File: docs/source/user_guide/backends/omnisci.rst [hunk 5]
```diff
@@ -109,8 +112,8 @@ expression. Its main methods are:
 
 `_build_ast` method is required.
 
-`MapDQuery` class should be used redefine at least `_fetch` method. If `Query`
-class is used instead, when `MapDClient.execute` method is called, a exception
+`MapDQuery` class should define at least the `_fetch` method. If `Query`
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:115]:
```suggestion
`OmniSciDBQuery` class should define at least the `_fetch` method. If `Query`
```

File: docs/source/user_guide/backends/omnisci.rst [hunk 6]
```diff
@@ -109,8 +112,8 @@ expression. Its main methods are:
...
+class is used when the `MapDClient.execute` method is called, an exception
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:116]:
```suggestion
class is used when the `OmniSciDBClient.execute` method is called, an exception
```

File: docs/source/user_guide/backends/omnisci.rst [hunk 7]
```diff
@@ -124,11 +127,12 @@ is raised.
 `MapDCursor` class was created just to allow `ibis.client.Query.execute`
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:127]: I would remove from L124 to L132 (https://github.com/ibis-project/ibis/pull/2265/files#diff-4ae26e229bda1f1f9f4fda583d639683L124-L132)

and I would add something like:

The `execution` method, under the hood, will use some `cursor` class that will convert the result from the database to some data frame format (e.g. Pandas dataframe, GeoPandas dataframe, cuDF dataframe).

File: docs/source/user_guide/backends/omnisci.rst [hunk 8]
```diff
@@ -124,11 +127,12 @@ is raised.
...
+.. code-block:: python
+
     with self.client._execute(self.compiled_ddl, results=True) as cur:
        ...
 
-Otherwise, `MapDQuery` should rewrites `execute` method with no `with`
+Otherwise, `MapDQuery` should implement the `execute` method with no `with`
 statement.
 
 compiler
```
Comment (xmnlab) [inline docs/source/user_guide/backends/omnisci.rst:138]: The list of the classes needed to be updated:

```
- OmniSciDBDialect
- OmniSciDBExprTranslator
- OmniSciDBQueryBuilder
- OmniSciDBSelect
- OmniSciDBSelectBuilder
- OmniSciDBTableSetFormatter
```

File: docs/source/user_guide/backends/omnisci.rst (hunk 4 above)
Comment (jreback) [reply docs/source/user_guide/backends/omnisci.rst:92]: are these listed in the api section? e.g. so we can reference them

**Comment by datapythonista** [comment]
Thanks for the suggestions @xmnlab, I applied the changes.

I added the Sphinx links to the classes and functions that are public and in the docs. Not sure if there are classes that should be in the docs and are not, but I'll leave that for another PR.

**Comment by jreback** [comment]
thanks @datapythonista and @xmnlab 

side issue, i bet the doc-build has warnings? we should fail on this.

**Comment by datapythonista** [comment]
> side issue, i bet the doc-build has warnings? we should fail on this.

Why do you think we've got warnings? We build the docs with `-W` which should turn warnings into errors: https://github.com/ibis-project/ibis/blob/master/ci/azure/linux.yml#L301

**Comment by jreback** [comment]
oh ok that's great then

**Pull request merged**